iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -98,101 +116,96 @@ private void ConnectionStatusChangeHandler(ConnectionStatus status, ConnectionSt
                     s_deviceClient.SetConnectionStatusChangesHandler(ConnectionStatusChangeHandler);
-                    s_deviceClient.OperationTimeoutInMilliseconds = (uint)s_operationTimeout.TotalMilliseconds;
+
                     _logger.LogDebug($"Initialized the client instance.");
                 }
+                s_initializeClientSemaphore.Release();
             }
+
+            // OpenAsync() is an idempotent call, it has the same effect if called once or multiple times on the same client.
+            await s_deviceClient.OpenAsync();
+            _logger.LogDebug($"Successfully opened the client instance.");
         }
 
         private async Task SendMessagesAsync(CancellationToken cancellationToken)
         {
             int count = 0;
             while (!cancellationToken.IsCancellationRequested)
             {
-                count++;
-                _logger.LogInformation($"Device sending message {count} to IoTHub...");
-
-                try
-                {
-                    await SendMessageAsync(count);
-                }
-                catch (Exception ex) when (ex is IotHubException exception && exception.IsTransient)
-                {
-                    // Inspect the exception to figure out if operation should be retried, or if user-input is required.
-                    _logger.LogError($"An IotHubException was caught, but will try to recover and retry explicitly: {ex}");
-                }
-                catch (Exception ex) when (ExceptionHelper.IsNetworkExceptionChain(ex))
+                // If the client is in Connected state, perform the desired operation.
+                if (s_connectionStatus == ConnectionStatus.Connected)
                 {
-                    _logger.LogError($"A network related exception was caught, but will try to recover and retry explicitly: {ex}");
-                }
+                    count++;
+                    _logger.LogInformation($"Device sending message {count} to IoTHub...");
+
+                    try
+                    {
+                        await SendMessageAsync(count);
+                    }
+                    catch (Exception ex) when (ex is IotHubException exception && exception.IsTransient)
+                    {
+                        // Inspect the exception to figure out if operation should be retried, or if user-input is required.
+                        _logger.LogError($"An IotHubException was caught, but will try to recover and retry explicitly: {ex}");
+                    }
+                    catch (Exception ex) when (ExceptionHelper.IsNetworkExceptionChain(ex))
+                    {
+                        _logger.LogError($"A network related exception was caught, but will try to recover and retry explicitly: {ex}");
+                    }
 
-                await Task.Delay(s_sleepDuration);
+                    await Task.Delay(s_sleepDuration);
```

Comment:
Do you want to delay only if connected? If not, this tight loop could eat up processor.

Reply:
yes, I am going to add some kind of delay or notifier event in here; looping continuously will not be efficient at all